fml/thread.cc

```diff
@@ -55,6 +72,17 @@ void Thread::SetCurrentThreadName(const std::string& name) {
   pthread_setname_np(name.c_str());
 #elif OS_LINUX || OS_ANDROID
   pthread_setname_np(pthread_self(), name.c_str());
+#elif OS_WIN
+  THREADNAME_INFO info;
+  info.dwType = 0x1000;
+  info.szName = name.c_str();
+  info.dwThreadID = GetCurrentThreadId();
+  info.dwFlags = 0;
+  __try {
```

Comment:
It is my understanding that __try is a Microsoft extension to deal with SEH (https://msdn.microsoft.com/en-us/library/windows/desktop/ms680657(v=vs.85).aspx). It's separate from C++'s exception handling. SEH handling should not be affected by disabling C++ exceptions.